src/bicep-registry/scheduled-action/main.bicep

```diff
@@ -0,0 +1,138 @@
+// @resourceGroup @subscription @tenant
+
+// See https://learn.microsoft.com/rest/api/cost-management/scheduled-actions/create-or-update
+
+//===| Parameters |============================================================
+
+targetScope = 'subscription'
+
+@description('Name of the scheduled action used in the resource ID.')
+param name string
+
+@description('Indicates the kind of scheduled action.')
+@allowed([
+  'Email'
+  'InsightAlert' // @subscription
+])
+param kind string = 'Email'
+
+// @tenant
+//   @description('The Azure resource ID for the billing scope to use. This is the full resource ID of a billing account, billing profile, department, etc. Do not use the GUID only.')
+//   param scope string
+
+@description('Specifies which built-in view to use. This is a shortcut for the full view ID.')
+@allowed([
+  ''
+  'AccumulatedCosts'
+  'DailyCosts'
+  'InvoiceDetails'
+])
+param builtInView string = ''
+
+@description('Required if kind is "Email" and builtInView is not set. The resource ID of the view to which the scheduled action will send. The view must either be private (tenant level) or owned by the same scope as the scheduled action. Ignored if kind is "InsightAlert" or if builtInView is set.')
+param viewId string = ''
```

Comment:
In order to make something optional, there has to be a default value. I'm using `''` here and we check that later.